File: crates/tldr-cli/tests/debt_sqale_fixtures/big.ex
# Elixir debt fixture for M-015 - complexity, nesting, long method, TODOs

defmodule Bigger do
  def extremely_complex_function(a, b, c, d, e, f, g) do
    # TODO: refactor this monster
    result = cond do
      a > 0 ->
        cond do
          b > 0 ->
            cond do
              c > 0 ->
                cond do
                  d > 0 ->
                    cond do
                      e > 0 ->
                        cond do
                          f > 0 -> a + b + c + d + e + f + g
                          f < 0 -> a - b
                          true -> 0
                        end
                      true -> -1
                    end
                  true -> -2
                end
              true -> -3
            end
          true -> -4
        end
      true -> -5
    end
    case a do
      1 -> result + 1
      2 -> result + 2
      3 -> result + 3
      4 -> result + 4
      5 -> result + 5
      6 -> result + 6
      7 -> result + 7
      8 -> result + 8
      9 -> result + 9
      10 -> result + 10
      _ -> result
    end
  end

  def another_long_method() do
    # FIXME: this should be split
    x = 0
    x = x + 1
    x = x + 2
    x = x + 3
    x = x + 4
    x = x + 5
    x = x + 6
    x = x + 7
    x = x + 8
    x = x + 9
    x = x + 10
    x = x + 11
    x = x + 12
    x = x + 13
    x = x + 14
    x = x + 15
    x = x + 16
    x = x + 17
    x = x + 18
    x = x + 19
    x = x + 20
    x = x + 21
    x = x + 22
    x = x + 23
    x = x + 24
    x = x + 25
    x = x + 26
    x = x + 27
    x = x + 28
    x = x + 29
    x = x + 30
    x = x + 31
    x = x + 32
    x = x + 33
    x = x + 34
    x = x + 35
    x = x + 36
    x = x + 37
    x = x + 38
    x = x + 39
    x = x + 40
    x = x + 41
    x = x + 42
    x = x + 43
    x = x + 44
    x = x + 45
    x = x + 46
    x = x + 47
    x = x + 48
    x = x + 49
    x = x + 50
    x = x + 51
    x = x + 52
    x = x + 53
    x = x + 54
    x
  end
end

File: crates/tldr-cli/tests/debt_sqale_fixtures/big.c
/* C debt fixture for M-015 - complexity, nesting, long method, TODOs */

int extremely_complex_function(int a, int b, int c, int d, int e, int f, int g) {
    /* TODO: refactor this monster */
    int result = 0;
    if (a > 0) {
        if (b > 0) {
            if (c > 0) {
                if (d > 0) {
                    if (e > 0) {
                        if (f > 0) {
                            result = a + b + c + d + e + f + g;
                        } else if (f < 0) {
                            result = a - b;
                        } else {
                            result = 0;
                        }
                    } else {
                        result = -1;
                    }
                } else {
                    result = -2;
                }
            } else {
                result = -3;
            }
        } else {
            result = -4;
        }
    } else {
        result = -5;
    }
    switch (a) {
        case 1: result += 1; break;
        case 2: result += 2; break;
        case 3: result += 3; break;
        case 4: result += 4; break;
        case 5: result += 5; break;
        case 6: result += 6; break;
        case 7: result += 7; break;
        case 8: result += 8; break;
        case 9: result += 9; break;
        case 10: result += 10; break;
    }
    return result;
}

int another_long_method(void) {
    /* FIXME: this should be split */
    int x = 0;
    x += 1;
    x += 2;
    x += 3;
    x += 4;
    x += 5;
    x += 6;
    x += 7;
    x += 8;
    x += 9;
    x += 10;
    x += 11;
    x += 12;
    x += 13;
    x += 14;
    x += 15;
    x += 16;
    x += 17;
    x += 18;
    x += 19;
    x += 20;
    x += 21;
    x += 22;
    x += 23;
    x += 24;
    x += 25;
    x += 26;
    x += 27;
    x += 28;
    x += 29;
    x += 30;
    x += 31;
    x += 32;
    x += 33;
    x += 34;
    x += 35;
    x += 36;
    x += 37;
    x += 38;
    x += 39;
    x += 40;
    x += 41;
    x += 42;
    x += 43;
    x += 44;
    x += 45;
    x += 46;
    x += 47;
    x += 48;
    x += 49;
    x += 50;
    x += 51;
    x += 52;
    x += 53;
    x += 54;
    return x;
}
